src/Network/NWDataTransferReport.cs

```diff
@@ -0,0 +1,174 @@
+//
+// NWDataTransferReport.cs: Bindings the Network nw_data_transfer_report_t API.
+//
+// Authors:
+//   Manuel de la Pena (mandel@microsoft.com)
+//
+// Copyrigh 2019 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using OS_nw_data_transfer_report=System.IntPtr;
+using OS_nw_connection=System.IntPtr;
+using OS_nw_interface=System.IntPtr;
+
+namespace Network {
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public enum NWDataTransferReportState {
+		Collecting = 1,
+		Collected = 2
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public class NWDataTransferReport : NativeObject {
+
+		public NWDataTransferReport (IntPtr handle, bool owns) : base (handle, owns) {}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_data_transfer_report nw_connection_create_new_data_transfer_report (OS_nw_connection connection);
+
+		public NWDataTransferReport (NWConnection connection)
+		{
+			if (connection == null)
+				throw new ArgumentNullException (nameof (connection));
+
+			InitializeHandle (nw_connection_create_new_data_transfer_report (connection.Handle));
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern uint nw_data_transfer_report_get_path_count (OS_nw_data_transfer_report report);
+
+		public uint PathCount => nw_data_transfer_report_get_path_count (GetCheckedHandle ());
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_duration_milliseconds (OS_nw_data_transfer_report report);
+
+		public ulong Duration => nw_data_transfer_report_get_duration_milliseconds (GetCheckedHandle ());
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_interface nw_data_transfer_report_copy_path_interface (OS_nw_data_transfer_report report, uint path_index);
+
+		public NWInterface GetInterface (uint pathIndex)
```

Comment:
All of these properties with `Get` should either be made methods or have the `Get` prefix removed. As a property, the `Get` is implicit in using the getter.

Reply:
ah! but they take a parameter! we are getting the interface in an index, all other methods that do not take a parameter (unless I missed one) are exposed as properties.

Reply:
@stephen-hawley I'm going to add and indexer, which is nice to have :)